storage/src/storage.rs

```diff
@@ -67,13 +68,16 @@ impl Storage {
                         .unwrap_or(u64::MAX);
 
                     let latest_snapshot = RwLock::new(Snapshot::new(db.clone(), jmt_version));
+                    let latest_snapshot_tmp = Snapshot::new(db.clone(), jmt_version);
 
                     // We discard the receiver here, because we'll construct new ones in subscribe()
                     let (snapshot_tx, _) =
                         watch::channel(StateNotification(latest_snapshot.read().clone()));
 
+                    let snapshots = RwLock::new(SnapshotCache::new(latest_snapshot_tmp, 10));
```

Comment:
Wait, why are we constructing an `RwLock<Snapshot>` here? Shouldn't this be something like
```rust
let snapshots = RwLock::new(SnapshotCache::new(
    Snapshot::new(db.clone(), jmt_version),
    10
    ));

let (snapshot_tx, _) = watch::channel(StateNotification(snapshots.read().latest()));
```
since we're not using an `RwLock`'d snapshot any more?